app/controllers/SeriesDetailsController.scala

```diff
@@ -19,33 +25,59 @@ class SeriesDetailsController @Inject()(val controllerComponents: SecurityCompon
                                         )(implicit val ec: ExecutionContext) extends TokenSecurity with I18nSupport {
 
   private val secureAction = Secure("OidcClient")
-  private val client = graphqlConfiguration.getClient[Data, Variables]()
+  private val getSeriesClient = graphqlConfiguration.getClient[getSeries.Data, getSeries.Variables]()
+  private val addConsignmentClient = graphqlConfiguration.getClient[addConsignment.Data, addConsignment.Variables]()
 
   val selectedSeriesForm = Form(
     mapping(
-      "seriesId" -> nonEmptyText
+      "series" -> nonEmptyText
     )(SelectedSeriesData.apply)(SelectedSeriesData.unapply)
   )
 
-  def seriesDetails(): Action[AnyContent] = secureAction.async { implicit request: Request[AnyContent] =>
-    val userTransferringBody = keycloakConfiguration.verifyToken(request.token.getValue)
-      .map(t => t.getOtherClaims.get("body").asInstanceOf[String])
-    val variables: Variables = new Variables(userTransferringBody)
+  private def getSeriesDetails(request: Request[AnyContent], status: Status, form: Form[SelectedSeriesData])(implicit requestHeader: RequestHeader) = {
+    val userTransferringBody: Option[String] = keycloakConfiguration.verifyToken(request.token.getValue)
+      .flatMap(t => t.getOtherClaims.asScala.get("body").asInstanceOf[Option[String]])
```

Comment:
`asInstanceOf[Option[String]]` looks much nicer than the empty string alternative, thanks. :+1: